app/graphql/mutations/create_project.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Mutations
+  # Base Mutation
+  class CreateProject < BaseMutation
+    null true
+    description 'Create a new project..'
+    argument :description, String, required: false, description: 'The description to give the project.'
+    argument :group_id, ID,
+             required: false,
+             description: 'The Node ID of a group to create the project in. For example, `gid://irida/Group/a84cd757-dedb-4c64-8b01-097020163077`.' # rubocop:disable Layout/LineLength
+    argument :group_puid, ID, # rubocop:disable GraphQL/ExtractInputType
+             required: false,
+             description: 'Persistent Unique Identifier of a group to create the project in. For example, `INXT_GRP_AAAAAAAAAA`.' # rubocop:disable Layout/LineLength
+    argument :name, String, required: true, description: 'The name to give the project.' # rubocop:disable GraphQL/ExtractInputType
+    argument :path, String, required: false, description: 'A custom path for the project.' # rubocop:disable GraphQL/ExtractInputType
+
+    field :errors, [Types::UserErrorType], null: false, description: 'A list of errors that prevented the mutation.'
+    field :project, Types::ProjectType, description: 'The newly created project.'
+
+    def resolve(args)
+      namespace = get_namespace(args)
+
+      if namespace.nil? || !namespace.persisted?
+        user_errors = [{ path: ['group'], message: 'Group not found by provided ID or PUID' }]
+        return { project: nil, errors: user_errors }
+      end
+
+      # slugify path, if no path given slugify name to use as path
+      args[:path] = if args[:path]
+                      args[:path].to_s.parameterize(separator: '-')
```

Comment:
**nitpick (blocking):** Can you only slugify `path` only if you have to generate it from `name` (e.g. path is not sent as a argument). Otherwise accept the `path` provided and allow that to raise an error if it does not pass validation.

Reply:
Fixed in https://github.com/phac-nml/irida-next/pull/795/commits/de833800d4766f5f65ef80f657f56cb00bc8505c